### DISCUSSION TITLE
[v5] auth setup cannot find '@ioc:Adonis/Addons/Auth'

### DISCUSSION THREAD

**Discussion by vookio** [adonisjs/core#2316] [Help] [answered]
Hello,
Probably I am doing something wrong since it would be already reported in case this was an issue. That's why I didn't create one. Could you please help me to figure out this?

I follow https://preview.adonisjs.com/guides/auth/setup to setup auth (api guard, lucid model).

In both config/auth.ts and contracts/auth.ts files, the message below is shown when I hover '@ ioc:Adonis/Addons/Auth' (space after @ intentionally left here)

```
import { AuthConfig } from '@ioc:Adonis/Addons/Auth'

import { LucidProviderContract, LucidProviderConfig, 
         OATGuardContract, OATGuardConfig } from '@ioc:Adonis/Addons/Auth'

>> Cannot find module '@ioc:Adonis/Addons/Auth' or its corresponding type declarations.
```
Also:
```
declare module '@ioc:Adonis/Addons/Auth'

>> Invalid module name in augmentation, module '@ioc:Adonis/Addons/Auth' cannot be found.
```
And "auth" not resolved in HttpContextContract (in a Controller)
```
public async login ({ request, _auth_ }: HttpContextContract) {
...
}

>> Property 'auth' does not exist on type 'HttpContextContract'.
```
Using versions:
```
"@adonisjs/auth": "^5.1.1",
"@adonisjs/core": "^5.0.4-preview-rc-2.1",
"@adonisjs/lucid": "^10.0.0",
```

**Comment by RomainLanz** [comment, marked as answer]
Hey @vookio! 👋 

Try to restart the TypeScript Server of your IDE (or directly your IDE).
Sometimes, the IDE doesn't load the typing well.

If that doesn't fix anything, could you please try to delete your `node_modules` folder and re-run `npm install`?

**Reply by vookio** [reply]
Restarting the IDE fixed it 🤦‍♂️ thanks! 👍

**Reply by og-vale** [reply]
I'm having the same issue. I've tried the above and solution and I'm still getting that error (Cannot find module '@ioc:Adonis/Addons/Auth' or its corresponding type declarations.ts)

**Reply by Maths032** [reply]
> Estou tendo o mesmo problema. Eu tentei a solução acima e ainda estou recebendo aquele erro (Não é possível encontrar o módulo ' @IOC : Adonis / Addons / Auth' ou seu tipo correspondente declarations.ts)

I really have a problem. do you solve your problem? Do you know the solution?

**Reply by vinicioslc** [reply]
After removing / installing node_modules again and restarting VSCode, intellisense was fixed

**Reply by pktippa** [reply]
for me it got fixed, after adding `"@adonisjs/auth"` to tsconfig.js -> compilerOptions -> types array

**Reply by RomainLanz** [reply]
> for me it got fixed, after adding `"@adonisjs/auth"` to tsconfig.js -> compilerOptions -> types array

This should be automatically done when running `node ace configure @adonisjs/auth`.
As state inside the documentation.

**Comment by robenson-dev** [comment]
use this command should resolve your problem : `pnpm create adonis-ts-app G-adonis`
